### PR TITLE
Expand Plasma developer docs

### PR DESCRIPTION
**Pull request by emmanuel-awosika**
Edited the page to add a detailed description of plasma and how it works. I also added new information on the mass exit problem and how it threatens plasma's security model.

- Expanded upon the current docs to more thoroughly describe Plasma
- Explained [the mass exit problem](https://www.google.com/search?q=plasma+mass+exit+problem)
- Explained why plasma fell out of favour and how the Ethereum community has gathered around other Layer 2 solutions.

## Related Issue
#3655

### CODE REVIEW

**Comment by emmanuel-awosika** [comment]
I'm not sure, but I think it's safe to take out the earlier section I added comparing plasma to sharding and sidechains. Besides concerns over the size of the page, I think a "plasma vs layer 2" comparison is more useful (and I already briefly touched on the sidechain vs plasma debate in this new edit).

File: src/content/developers/docs/scaling/plasma/index.md [hunk 1]
```diff
@@ -6,22 +6,137 @@ sidebar: true
...
+
+### Off-chain computation {#off-chain-computation}
+
+Ethereum's current processing speed is limited to ~ 15-20 transactions per second, reducing the short-term possibility of scaling to handle more users. This problem exists mainly because Ethereum's [consensus mechanism](/developers/docs/consensus-mechanisms/) requires many peer-to-peer nodes to verify every update to the blockchain's state.  
+
+Although Ethereum's consensus mechanism is necessary for security, it may not apply to every use case. For example, Alice may not need her daily payments to Bob for a cup of coffee verified by the entire Ethereum network since some trust exists between both parties. 
+
+Plasma supposes that Ethereum Mainnet doesn't need to verify all transactions. Instead, we can process transactions off Mainnet, freeing nodes from having to validate every transaction. 
+
+Off-chain computation is necessary since Plasma chains can optimize for speed and cost. For example, a Plasma chain may—and most often does—use a single "Operator" to manage the ordering and execution of transactions. With just one entity verifying transactions, processing times on a plasma chain are faster than Ethereum Mainnet.
```
[the remark this answers is not in the document]
Comment (emmanuel-awosika) [reply src/content/developers/docs/scaling/plasma/index.md:35]: Not really. The proper noun was for emphasis (i.e., to show the operator is a centralized entity controlling the chain). I'll revise it now, though.

File: src/content/developers/docs/scaling/plasma/index.md [hunk 2]
```diff
@@ -6,22 +6,137 @@ sidebar: true
...
 
-A plasma chain is a separate blockchain that is anchored to the main Ethereum chain, and uses fraud proofs (like [optimistic rollups](/developers/docs/scaling/optimistic-rollups/)) to arbitrate disputes. These chains are sometimes referred to as "child" chains as they are essentially smaller copies of the Ethereum Mainnet. Merkle trees enable creation of a limitless stack of these chains that can work to offload bandwidth from the parent chains (including Mainnet). These derive their security through [fraud proofs](/glossary/#fraud-proof), and each child chain has its own mechanism for block validation.
+Merkle trees enable the creation of a limitless stack of these chains that can work to offload bandwidth from parent chains (including Ethereum Mainnet). However, while these chains derive some security from Ethereum (via fraud proofs), their security and efficiency are affected by several design limitations.
```
Comment (minimalsm) [inline src/content/developers/docs/scaling/plasma/index.md:11]:
```suggestion
Merkle trees enable the creation of an endless stack of these chains that can work to offload bandwidth from parent chains (including Ethereum Mainnet). However, while these chains derive some security from Ethereum (via fraud proofs), their security and efficiency are affected by several design limitations.  
```

File: src/content/developers/docs/scaling/plasma/index.md [hunk 3]
```diff
@@ -6,22 +6,137 @@ sidebar: true
...
+You should have a good understanding of all the foundational topics and a high-level understanding of [Ethereum scaling](/developers/docs/scaling/).
 
-## Pros and cons {#pros-and-cons}
+## What is Plasma?
+
+Plasma is a framework for improving scalability in public blockchains like Ethereum. As described in the original [Plasma whitepaper](http://plasma.io/plasma.pdf), Plasma chains are built atop another blockchain (called a "root chain"). Each "child chain" extends from the root chain and is generally managed by a smart contract deployed on the parent chain. 
+
+The Plasma contract functions, among other things, as a [bridge](/developers/docs/bridges/), allowing users to move assets between Ethereum Mainnet and the plasma chain. Although this makes them similar to [sidechains](/developers/docs/scaling/sidechains/), plasma chains benefit—at least, to some extent—from Ethereum Mainnet's security. This is unlike sidechains that are solely responsible for their security.
```
Comment (minimalsm) [inline src/content/developers/docs/scaling/plasma/index.md:21]:
```suggestion
The Plasma contract functions, among other things, as a [bridge](/developers/docs/bridges/) allowing users to move assets between Ethereum Mainnet and the plasma chain. Although this makes them similar to [sidechains](/developers/docs/scaling/sidechains/), plasma chains benefit—at least, to some extent—from Ethereum Mainnet's security. This is unlike sidechains that are solely responsible for their security. 
```

**Review by minimalsm — APPROVED** [review]
This looks great to me :-) I don't have a comprehensive understanding of Plasma at a technical level so I can't comment on tiny nuances here but we're struggling to get reviews from the people who do, so I'm merging and I'm sure the internet will crawl over glass to tell us we're wrong in the places we're wrong 😁 awesome work as always @emmanuel-awosika

**Comment by emmanuel-awosika** [comment]
> This looks great to me :-) I don't have a comprehensive understanding of Plasma at a technical level so I can't comment on tiny nuances here but we're struggling to get reviews from the people who do, so I'm merging and I'm sure the internet will crawl over glass to tell us we're wrong in the places we're wrong 😁 awesome work as always @emmanuel-awosika

Yeah, that works. I like the idea of relying on the "wisdom of the crowds", too.

**Pull request merged**